### DISCUSSION TITLE
Auto-completion problem when using fish shell

### DISCUSSION THREAD

**Discussion by otowa1** [Homebrew/discussions#4332] [Linux]
### Additional Info

- [X] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0c1031f2fd71016f0f5df699caaed384487bfb16
Last commit: 7 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: bc18192d5f01cfc11a569af66792fb5009e42674
Core tap last commit: 3 weeks ago
Core tap branch: master
Core tap JSON: 17 Mar 06:08 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: quad-core 64-bit skylake
Clang: N/A
Git: 2.25.1 => /bin/git
Curl: 7.68.0 => /bin/curl
Kernel: Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 GNU/Linux
OS: Unknown
WSL: 2 (Microsoft Store)
Host glibc: 2.31
/usr/bin/gcc: 9.4.0
/usr/bin/ruby: N/A
==> Downloading https://formulae.brew.sh/api/formula.jws.json
######################################################################## 100.0%
glibc: 2.35_1
gcc@11: N/A
gcc: 12.2.0
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Hi,
In my environment, the variable $fish_complete_path used for fish shell auto-completion did not contain a brew path, so I set the path according to the following manual.
https://docs.brew.sh/Shell-Completion

However, this would cause the brew path to be read twice if I reloaded config.fish with `"source ~/.config/fish/config.fish"`. 
`/home/user/.config/fish/completions /etc/fish/completions /home/user/.local/share/fish/vendor_completions.d /usr/share/fish/vendor_completions.d /usr/share/fish/completions /home/user/.local/share/fish/generated_completions /home/linuxbrew/.linuxbrew/share/fish/vendor_completions.d /home/linuxbrew/.linuxbrew/share/fish/vendor_completions.d`
In that case, completion does not seem to work. 

I can think of a way to do the following, for example
```
function checkCompletions
    for element in $argv
        if test $element = $argv[1]
            return 0
        end
    end
    return 1
end

if test -d (brew --prefix)"/share/fish/completions"
    checkCompletions $fish_complete_path (brew --prefix)/share/fish/completions; or set -gx fish_complete_path $fish_complete_path (brew --prefix)/share/fish/completions
end
if test -d (brew --prefix)"/share/fish/vendor_completions.d"
    checkCompletions $fish_complete_path (brew --prefix)/share/fish/vendor_completions.d; or set -gx fish_complete_path $fish_complete_path (brew --prefix)/share/fish/vendor_completions.d
end
```
I don't think it is a very elegant solution.I feel I don't understand the issue.Is this a known problem and is this the only way to do it?

**Comment by carlocab** [comment]
How did you install `fish`?

**Reply by otowa1** [reply]
I used apt.

**Reply by carlocab** [reply]
You can avoid it by installing `fish` from Homebrew.

Otherwise -- it's up to you how to make your `fish.config` idempotent. The solution you have seems fine.

**Reply by otowa1** [reply]
I see. I will refer to this the next time I install Linux for a development environment (which I probably will have the opportunity to do). Thank you.
